File: system-test/java/src/main/java/com/optivem/shop/testkit/core/usecase/external/tax/usecases/GetTaxRate.java
package com.optivem.shop.testkit.core.usecase.external.tax.usecases;

import com.optivem.shop.testkit.core.shared.UseCaseContext;
import com.optivem.shop.testkit.core.shared.UseCaseResult;
import com.optivem.shop.testkit.core.usecase.external.tax.usecases.base.BaseTaxUseCase;
import com.optivem.shop.testkit.driver.port.external.tax.TaxDriver;
import com.optivem.shop.testkit.driver.port.external.tax.dtos.GetTaxResponse;

public class GetTaxRate extends BaseTaxUseCase<GetTaxResponse, GetTaxVerification> {
    private String countryValueOrAlias;

    public GetTaxRate(TaxDriver driver, UseCaseContext context) {
        super(driver, context);
    }

    public GetTaxRate country(String countryValueOrAlias) {
        this.countryValueOrAlias = countryValueOrAlias;
        return this;
    }

    @Override
    public UseCaseResult<GetTaxResponse, GetTaxVerification> execute() {
        var country = context.getParamValueOrLiteral(countryValueOrAlias);

        var result = driver.getTaxRate(country);

        return new UseCaseResult<>(result, context, GetTaxVerification::new);
    }
}

File: system-test/dotnet/Dsl.Core/UseCase/External/Tax/UseCases/GetTaxRate.cs
using Driver.Port.External.Tax;
using Driver.Port.External.Tax.Dtos;
using Driver.Port.External.Tax.Dtos.Error;
using Dsl.Core.External.Tax.UseCases.Base;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Tax.UseCases;

public class GetTaxRate : BaseTaxUseCase<GetTaxResponse, GetTaxVerification>
{
    private string? country;

    public GetTaxRate(ITaxDriver driver, UseCaseContext context)
        : base(driver, context)
    {
    }

    public GetTaxRate Country(string? country)
    {
        this.country = country;
        return this;
    }

    public override async Task<TaxUseCaseResult<GetTaxResponse, GetTaxVerification>> Execute()
    {
        var countryValue = _context.GetParamValueOrLiteral(country);

        var result = await _driver.GetTaxRateAsync(countryValue);

        return new TaxUseCaseResult<GetTaxResponse, GetTaxVerification>(
            result,
            _context,
            (response, ctx) => new GetTaxVerification(response, ctx));
    }
}
